packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -340,3 +356,294 @@ export function powerMetricWeeks(
+  const metricMean = getMetricMean(metric);
+  const priorStandardDeviationDGP = 0;
+  return calculatePriorVariance(
+    Math.pow(priorStandardDeviationDGP, 2),
+    metricMean,
+    relative
+  );
+}
+
+// Function to estimate variance of tau hat conditional on tau
+function estimateTauHatVariance(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const s = powerStandardError(metric, nPerVariation, relative);
+  return Math.pow(s, 2);
+}
+
+// Function to calculate marginal variance of tau hat
+function getMarginalVarianceTauHat(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const priorVarianceDGP = calculatePriorVarianceDGP(metric, relative);
+  const tauHatVariance = estimateTauHatVariance(
+    metric,
+    nPerVariation,
+    relative
+  );
+  return tauHatVariance + priorVarianceDGP;
+}
+
+// Function to calculate posterior precision
+function getPosteriorPrecision(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const priorVarianceSpecified = calculatePriorVarianceSpecified(
+    metric,
+    relative
+  );
+  const tauHatVariance = estimateTauHatVariance(
+    metric,
+    nPerVariation,
+    relative
+  );
+  const properInt = metric.proper ? 1 : 0;
+  return 1 / tauHatVariance + properInt / priorVarianceSpecified;
+}
+
+// Function to calculate upper cutpoint
+export function getCutpoint(
+  metric: MetricParams,
+  alpha: number,
+  nPerVariation: number,
+  relative: boolean,
+  upper: boolean
+): number {
+  const priorMeanSpecified = calculatePriorMeanSpecified(metric, relative);
+  const priorVarianceSpecified = calculatePriorVarianceSpecified(
+    metric,
+    relative
+  );
+  const priorMeanDGP = calculatePriorMeanDGP(metric, relative);
+  const tauHatVariance = estimateTauHatVariance(
+    metric,
+    nPerVariation,
+    relative
+  );
+  const posteriorPrecision = getPosteriorPrecision(
+    metric,
+    nPerVariation,
+    relative
+  );
+  const marginalVarianceTauHat = getMarginalVarianceTauHat(
+    metric,
+    nPerVariation,
+    relative
+  );
+  const zStar = normal.quantile(1.0 - 0.5 * alpha, 0, 1);
+  const upperSign = upper ? 1 : -1;
+  const properInt = metric.proper ? 1 : 0;
+
+  const numerator =
+    upperSign * tauHatVariance * Math.sqrt(posteriorPrecision) * zStar -
+    (properInt * (tauHatVariance * priorMeanSpecified)) /
+      priorVarianceSpecified -
+    priorMeanDGP;
+  const denominator = Math.sqrt(marginalVarianceTauHat);
+  return numerator / denominator;
+}
+
+/**
+ * Performs power calculation
+ *
+ * @param metric an object of class MetricParams that has info about prior mean and sd, metric mean and sd, and effect size.
+ * @param alpha false positive rate (default: 0.05).
+ * @param nPerVariation sample size per variation.
+ * @param relative boolean indicator if relative inference is desired.
+ * @returns Estimated power.
+ */
+export function powerEstBayesian(
```

Comment:
This method and getCutpoint are really clear. Thanks!

Reply:
thanks!